Raise an error if batch_size cannot be inferred from current batch (#10541)
PyTorchLightning/pytorch-lightning: CHANGELOG.md

- Moved ownership of the `PrecisionPlugin` into `TrainingTypePlugin` and updated all references ([#10570](https://github.com/PyTorchLightning/pytorch-lightning/pull/10570))

> - Raised an error if the `batch_size` cannot be inferred from current batch incase if its a string or some custom batch ([#10541](https://github.com/PyTorchLightning/pytorch-lightning/pull/10541))

Reviewer: ```suggestion
- Raised an error if the `batch_size` cannot be inferred from the current batch if it contained a string or was a custom batch object ([#10541](https://github.com/PyTorchLightning/pytorch-lightning/pull/10541))
```
